tools/test/test_test_selections.py

```diff
@@ -1,7 +1,7 @@
 import random
 import unittest
 
-from tools.testing.test_selections import calculate_shards
+from tools.testing.test_selections import _calculate_shards
```

Comment:
Adding underscore in front of function name means it not supposed to be imported,  see https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables
```suggestion
from tools.testing.test_selections import calculate_shards
```